pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -85,22 +130,18 @@ func (pl *NodeAffinity) Score(ctx context.Context, state *framework.CycleState,
 		return 0, framework.AsStatus(fmt.Errorf("getting node %q from Snapshot: %w", nodeName, err))
 	}
 
-	affinity := pod.Spec.Affinity
-
 	var count int64
 	if pl.addedPrefSchedTerms != nil {
 		count += pl.addedPrefSchedTerms.Score(node)
 	}
-	// A nil element of PreferredDuringSchedulingIgnoredDuringExecution matches no objects.
-	// An element of PreferredDuringSchedulingIgnoredDuringExecution that refers to an
-	// empty PreferredSchedulingTerm matches all objects.
-	if affinity != nil && affinity.NodeAffinity != nil && affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution != nil {
-		// TODO(#96164): Do this in PreScore to avoid computing it for all nodes.
-		preferredNodeAffinity, err := nodeaffinity.NewPreferredSchedulingTerms(affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution)
-		if err != nil {
-			return 0, framework.AsStatus(err)
-		}
-		count += preferredNodeAffinity.Score(node)
+
+	s, err := getPreScoreState(state)
+	if err != nil {
```

Comment:
There might be users of KubeSchedulerConfig that have `disabled: ["*"]` for PreScore. Those users would start seeing failures without them doing changes. That's not desirable for a beta API.

Make it so that, if prescore state is not found, it gets calculated here. Of course, avoid code duplication and add a test.